sql/src/main/java/io/crate/planner/SelectStatementPlanner.java

```diff
@@ -59,7 +59,7 @@ public Visitor(LogicalPlanner logicalPlanner) {
         }
 
         private Plan invokeLogicalPlanner(QueriedRelation relation, Planner.Context context) {
-            Plan plan = logicalPlanner.plan(relation, context, FetchMode.WITH_PROPAGATION);
+            Plan plan = logicalPlanner.plan(relation, context, FetchMode.MAYBE_CLEAR);
```

Comment:
as afaik MAYBE_CLEAR is only used once, does it maybe make sense to create a `plan(...)` method with a default NEVER_CLEAR mode?

Reply:
`MAYBE_CLEAR` can also be used to plan subRelations. I'm in favor of keeping this explicit.